rpcs3/Emu/Cell/Modules/cellAtracXdec.cpp

```diff
@@ -270,35 +270,75 @@ void AtracXdecContext::exec(ppu_thread& ppu)
 		decoder.init_avcodec();
 	}
 
+	switch (savestate)
+	{
+	case atracxdec_state::waiting_for_cmd_mutex_lock: break;
+	case atracxdec_state::waiting_for_cmd_cond_wait: goto wait_for_cmd;
```

Comment:
We don't use goto in rpcs3

Reply:
I approve it in this case, 

I suggest mentioning where the label literally is:

```suggestion
	case atracxdec_state::waiting_for_cmd_cond_wait: goto wait_for_cmd_line_305;
```

Reply:
Just because you approve it personally doesn't make it ok

Reply:
He would have to replace it with a huge switch with fallthroughs for each case otherwise, this is not a refactoring pr but a bugfix so I rather let it be fixed first.

Reply:
In my opinion gotos is the most intuitive way of handling savestates here. We're simply jumping back to where the thread was interrupted, there aren't any crazy control flow gymnastics.
But if no gotos is a hard requirement, then I'll replace them with a large switch. I'd have to replace the command type switch with an if-else chain though since you can't put cases of one switch in the middle of another.
I disagree on putting the line number on the labels, however. You would have to update them every time the line number changes.

Reply:
Maybe not a line number, but I do want unique distinguishing on labels. Perhaps:

```suggestion
	case atracxdec_state::waiting_for_cmd_cond_wait: goto label2_wait_for_cmd_stage;
```

Putting the prefix "label" and relative order number between other labels in the function.
I do want to standardize it for the use of savestates.